source/DataNode.cpp

```diff
@@ -286,41 +287,39 @@ list<DataNode>::const_iterator DataNode::end() const noexcept
 
 
 
-// Print a message followed by a "trace" of this node and its parents.
-int DataNode::PrintTrace(const string &message) const
+// Recrusively print a message followed by a "trace" of this node and its parents.
```

Comment:
Is there any particular reason why a recursive method was chosen? Does it need to be recursive?

Reply:
I think it just produced nicer code.